packages/salesforcedx-vscode-core/src/commands/forceApexTestRun.ts

```diff
@@ -21,47 +22,86 @@ import {
   SfdxWorkspaceChecker
 } from './commands';
 
-class TestsSelector implements ParametersGatherer<vscode.QuickPickItem> {
+export enum TestType {
+  All,
+  Suite,
+  Class
+}
+
+export interface ApexTestQuickPickItem extends vscode.QuickPickItem {
+  type: TestType;
+}
+
+export class TestsSelector
+  implements ParametersGatherer<ApexTestQuickPickItem> {
   public async gather(): Promise<
-    CancelResponse | ContinueResponse<vscode.QuickPickItem>
+    CancelResponse | ContinueResponse<ApexTestQuickPickItem>
   > {
-    const files = await vscode.workspace.findFiles('**/*.testSuite-meta.xml');
-    const fileItems = files.map(file => {
+    const testSuites = await vscode.workspace.findFiles(
+      '**/*.testSuite-meta.xml'
+    );
+    const fileItems = testSuites.map(testSuite => {
       return {
         label: path
-          .basename(file.toString())
+          .basename(testSuite.toString())
           .replace('.testSuite-meta.xml', ''),
-        description: file.fsPath
+        description: testSuite.fsPath,
+        type: TestType.Suite
       };
     });
 
+    const apexClasses = await vscode.workspace.findFiles('**/*.cls');
+    apexClasses.forEach(apexClass => {
+      const fileContent = fs.readFileSync(apexClass.fsPath).toString();
+      if (fileContent && fileContent.toLowerCase().includes('@istest')) {
```

Comment:
Find files with .cls extension and filter the ones with "@istest" in the file.

Reply:
This is where we could use the apex language server in the future.